test/framework/cluster.go

```diff
@@ -1446,6 +1446,151 @@ func (e *ClusterE2ETest) VerifyPrometheusPackageInstalled(packageName string, ta
+		okCh <- "completed"
+	}()
+
+	select {
+	case err := <-errCh:
+		e.T.Fatal(err)
+	case <-okCh:
+		return
+	}
+}
+
+//go:embed testdata/certmanager/certmanager_selfsignedissuer.yaml
+var certManagerSelfSignedIssuer []byte
+
+//go:embed testdata/certmanager/certmanager_selfsignedcert.yaml
+var certManagerSelfSignedCert []byte
+
+func (e *ClusterE2ETest) verifySelfSignedCertificate(mgmtCluster *types.Cluster) error {
+	ctx := context.Background()
+	selfsignedCert := "my-selfsigned-ca"
+	err := e.KubectlClient.ApplyKubeSpecFromBytes(ctx, e.Cluster(), certManagerSelfSignedIssuer)
+	if err != nil {
+		return fmt.Errorf("error installing Cluster issuer for cert manager: %v", err)
+	}
+
+	err = e.KubectlClient.ApplyKubeSpecFromBytes(ctx, e.Cluster(), certManagerSelfSignedCert)
+	if err != nil {
+		return fmt.Errorf("error applying certificate for cert manager: %v", err)
+	}
+
+	err = e.KubectlClient.WaitJSONPathLoop(ctx, e.Cluster().KubeconfigFile, "5m", "status.conditions[0].status", "True",
+		fmt.Sprintf("certificates.cert-manager.io/%s", selfsignedCert), constants.EksaPackagesName)
+	if err != nil {
+		return fmt.Errorf("failed to issue a self signed certificate: %v", err)
+	}
+	return nil
+}
+
+//go:embed testdata/certmanager/certmanager_letsencrypt_issuer.yaml
+var certManagerLetsEncryptIssuer string
+
+//go:embed testdata/certmanager/certmanager_letsencrypt_cert.yaml
+var certManagerLetsEncryptCert []byte
+
+//go:embed testdata/certmanager/certmanager_secret.yaml
+var certManagerSecret string
+
+func (e *ClusterE2ETest) verifyLetsEncryptCert(mgmtCluster *types.Cluster) error {
+	ctx := context.Background()
+	letsEncryptCert := "test-cert"
+	accessKey, secretAccess, region, zoneID := GetRoute53Configs()
+	data := map[string]interface{}{
+		"route53SecretAccessKey": secretAccess,
+	}
+
+	certManagerSecretData, err := templater.Execute(certManagerSecret, data)
+	if err != nil {
+		return fmt.Errorf("failed creating cert manager secret: %v", err)
+	}
+
+	err = e.KubectlClient.ApplyKubeSpecFromBytes(ctx, e.Cluster(), certManagerSecretData)
+	if err != nil {
+		return fmt.Errorf("error creating cert manager secret: %v", err)
+	}
+
+	data = map[string]interface{}{
+		"route53AccessKeyId": accessKey,
+		"route53ZoneId":      zoneID,
+		"route53Region":      region,
+	}
+
+	certManagerIssuerData, err := templater.Execute(certManagerLetsEncryptIssuer, data)
+	if err != nil {
+		return fmt.Errorf("failed creating lets encrypt issuer: %v", err)
+	}
+
+	err = e.KubectlClient.ApplyKubeSpecFromBytes(ctx, e.Cluster(), certManagerIssuerData)
+	if err != nil {
+		return fmt.Errorf("error creating cert manager let's encrypt issuer: %v", err)
+	}
+
+	err = e.KubectlClient.ApplyKubeSpecFromBytes(ctx, e.Cluster(), certManagerLetsEncryptCert)
+	if err != nil {
+		return fmt.Errorf("error creating cert manager let's encrypt issuer: %v", err)
+	}
+
+	err = e.KubectlClient.WaitJSONPathLoop(ctx, e.Cluster().KubeconfigFile, "5m", "status.conditions[0].status", "True",
```

Comment:
This will succeed even if the certificate is in issuing state.

Reply:
That has not been my experience. I have generally seen that this status is False when the certificate is not ready. Do you know a better way of checking than checking the status conditions.

Reply:
https://github.com/cert-manager/cert-manager/blob/b5d1b1e27467e7fd9be4124f0041ec58524105c4/pkg/apis/certmanager/v1/types_certificate.go#L457
You must check that the type of the condition is `"Ready"` as well as status being `True`. It can also be in `"Issuing"` and status `True`, for example: https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/certificates/trigger/trigger_controller.go#L204